### PR TITLE
Bump IOGX v3

### PR DESCRIPTION
**Pull request by zeme-wana**
Changes to the interface:

1. `./nix/iogx-config.nix` renamed to -> `./nix/haskell.nix`
    - `iogx-config.nix:haskellCompilers` -> `./nix/haskell.nix:supportedCompilers`
    - `iogx-config.nix:defaultHaskellCompiler` -> `./nix/haskell.nix:defaultCompiler`
    - `iogx-config.nix:shouldCrossCompile` -> `./nix/haskell.nix:enableCrossCompilation`

2. `./nix/haskell-project.nix` renamed to -> `./nix/cabal-project.nix`

3. `./nix/pre-commit-check.nix` renamed to -> `./nix/formatters.nix`

4. `./nix/hydra-jobs.nix` renamed to -> `./nix/ci.nix`

5. `inputs` and `inputs'` have been swapped.
  `inputs` (no prime `'`) is now the original `inputs`, non-desystemised
  `inputs'` (with prime `'`) is now the de-systemized `inputs`
Reason behind this is that `inputs` is universally understood to be the original, unmodified flake inputs.
While `inputs'` is "non-standard" and thus warrants a different name.
This is the same notation used by [flake.parts](https://flake.parts/module-arguments#inputs)

6. The flake exports aliases for your haskell *executables* only in `outputs.apps`.
   For example, `outputs.apps.marlowe-apps-exe-marlowe-finder-ghc8107` also has an 
   alias `outputs.apps.marlowe-finder` (the cabal package name and the `exe` infix have been stripped)
   These can be access as usual with `inputs'.self.apps.marlowe-finder`

### CODE REVIEW

**Comment by zeme-wana** [comment]
The required statues have changed. From
```
ci/hydra-build:required.x86_64-darwin
ci/hydra-build:required.x86_64-linux
```
To
```
ci/hydra-build:x86_64-linux.required
ci/hydra-build:x86_64-darwin.required
```
You can check CI status here: https://ci.iog.io/jobset/input-output-hk-marconi/pullrequest-128
It's currently green.
Once merged, I'll update the required statuses in the Settings

File: nix/shell.nix [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
+  cardano-node = inputs'.cardano-node.legacyPackages.cardano-node;
 in
 {
   name = "marconi";
 
   packages = [
     cardano-cli
     cardano-node
-    inputs.mithril.packages.${pkgs.system}.mithril-client
+    inputs'.mithril.packages.mithril-client
```
Comment (kayvank) [inline nix/shell.nix:16]: :heart:

File: nix/scripts/start-benchmarking-machine.nix [hunk 2]
```diff
@@ -171,7 +172,7 @@ let
     };
 
     imports = [
-      inputs.cardano-node.nixosModules.cardano-node
+      inputs'.cardano-node.nixosModules.cardano-node
```
Comment (kayvank) [inline nix/scripts/start-benchmarking-machine.nix:175]: :+1:

File: nix/per-system-outputs.nix [hunk 3]
```diff
@@ -1,10 +0,0 @@
-# This file is part of the IOGX template and is documented at the link below:
-# https://www.github.com/input-output-hk/iogx#35-nixper-system-outputsnix
-
-{ inputs, inputs', pkgs, projects }:
-
-{
-
-  packages.marconi-sidechain = inputs.self.packages.marconi-sidechain-exe-marconi-sidechain-ghc927;
```
Comment (kayvank) [inline nix/per-system-outputs.nix:8]: the alias is not longer needed?

**Review by kayvank — APPROVED** [review]
LGTM

File: nix/cabal-project.nix [hunk 4]
```diff
@@ -1,13 +1,10 @@
 # This file is part of the IOGX template and is documented at the link below:
-# https://www.github.com/input-output-hk/iogx#33-nixhaskell-projectnix
+# https://www.github.com/input-output-hk/iogx#33-nixcabal-projectnix
 
-{ inputs, inputs', meta, config, pkgs, lib }:
+{ inputs, inputs', meta, config, pkgs, l, ... }:
```
Comment (koslambrou) [inline nix/cabal-project.nix:4]: Why not keep it "lib"? Seems clearer.

File: nix/haskell.nix [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# This file is part of the IOGX template and is documented at the link below:
+# https://www.github.com/input-output-hk/iogx#32-nixhaskellnix
+
+{
+  supportedCompilers = [ "ghc928" ];
+
+  enableCrossCompilation = true;
+
+  defaultChangelogPackages = [
+    "cardano-straming"
```
Comment (koslambrou) [inline nix/haskell.nix:10]: typo

File: nix/cabal-project.nix (hunk 4 above)
Comment (zeme-wana) [reply nix/cabal-project.nix:4]: `lib` is used so frequently that I thought it deserved a single-letter name. 
One can always do this instead:
```
import (pkgs) lib;
```

Comment (koslambrou) [reply nix/cabal-project.nix:4]: Yes `lib` is used frequently, but I've never seen `l`, so I thought *that* was confusing.

File: nix/cabal-project.nix [hunk 6]
```diff
@@ -77,14 +74,14 @@ let
...
-      export CARDANO_NODE=${inputs.cardano-node.legacyPackages.cardano-node}/bin/cardano-node${pkgs.stdenv.hostPlatform.extensions.executable}
+      export CARDANO_CLI=${inputs'.cardano-node.legacyPackages.cardano-cli}/bin/cardano-cli${pkgs.stdenv.hostPlatform.extensions.executable}
+      export CARDANO_NODE=${inputs'.cardano-node.legacyPackages.cardano-node}/bin/cardano-node${pkgs.stdenv.hostPlatform.extensions.executable}
       export CARDANO_NODE_SRC=${../.}
     ";
 
     # Needed for running the marconi-sidechain integration tests in CI
     marconi-sidechain.preCheck = "
-      export MARCONI_SIDECHAIN=${inputs.self.packages.marconi-sidechain-exe-marconi-sidechain-ghc927}/bin/marconi-sidechain
+      export MARCONI_SIDECHAIN=${inputs.self.packages.marconi-sidechain}/bin/marconi-sidechain
```
Comment (zeme-wana) [inline nix/cabal-project.nix:84]: Aliases for your cabal executables only (this excludes benchmarks and test suites) are added to your packages and apps flake outputs.
Alias means just the executable name as found in your cabal file, without the `$CabalPackageName-exe-` prefix.

For projects supporting a single compiler (like in this repo, ghc927), the compiler name is no longer appended to the flake outputs names.

inputs'.self.packages.marconi-sidechain-exe-marconi-sidechain-ghc927 ->
inputs'.self.packages.marconi-sidechain

inputs'.self.apps.marconi-sidechain-exe-marconi-sidechain-ghc927 ->
inputs'.self.apps.marconi-sidechain
```
If you cabal project has multiple executables with the same name across your cabal files, a warning will be generated when running `list-flake-outputs` inside `nix develop`, and no aliases will be created.

Comment (zeme-wana) [reply nix/cabal-project.nix]: FYI @koslambrou @kayvank

**Comment by zeme-wana** [comment]
CI is actually green due to hydra weirdness. I am merging now and updating the "required statues" names in the settings.

**Pull request merged**